lib/src/over_react_redux/over_react_flux.dart

```diff
@@ -0,0 +1,503 @@
+///
+/// UiFactory<CounterProps> ConnectedCounter = connectFlux<SmallCounterFluxStore, FluxActions, CounterProps>(
+///   mapStateToProps: (state) => (Counter()..count = state.count)
+/// )(Counter);
+///
+/// UiFactory<CounterProps> ConnectedBigCounter = connect<BigCounterFluxStore, FluxActions, CounterProps>(
+///   mapStateToProps: (state) => (Counter()..count = state.bigCount),
+///   context: bigCounterContext,
+/// )(Counter);
+///
+/// react_dom.render(
+///   Dom.div()(
+///     (ReduxProvider()..store = store1)(
+///       (ReduxProvider()
+///         ..store = store2
+///         ..context = bigCounterContext
+///       )(
+///         Dom.div()(
+///           Dom.h3()('ConnectedBigCounter Store2'),
+///           ConnectedBigCounter()(
+///             Dom.h4()('ConnectedCounter Store1'),
+///             ConnectedCounter()(),
+///           ),
+///         ),
+///       ),
+///     ),
+///   ), querySelector('#content')
+/// );
+/// ```
+///
+/// - [pure] if `true` (default), connect performs several equality checks that are used to avoid unnecessary
+/// calls to [mapStateToProps], [mapActionsToProps], [mergeProps], and ultimately to `render`. These include
+/// [areOwnPropsEqual], [areStatePropsEqual], and [areMergedPropsEqual].
+/// While the defaults are probably appropriate 99% of the time, you may wish to override them with custom
+/// implementations for performance or other reasons.
+///
+/// - [forwardRef] if `true`, the `ref` prop provided to the connected component will be return the wrapped component.
+///
+/// For more info see the <https://github.com/Workiva/over_react/blob/master/doc/flux_to_redux.md>.
+UiFactory<TProps> Function(UiFactory<TProps>) connectFlux<TStore extends flux.Store, TActions, TProps extends UiProps>({
+  Map Function(TStore state) mapStateToProps,
+  Map Function(TStore state, TProps ownProps) mapStateToPropsWithOwnProps,
+  Map Function(TActions actions) mapActionsToProps,
+  Map Function(TActions actions, TProps ownProps) mapActionsToPropsWithOwnProps,
+  Map Function(TProps stateProps, TProps dispatchProps, TProps ownProps) mergeProps,
+  bool Function(TProps nextProps, TProps prevProps) areOwnPropsEqual,
+  bool Function(TProps nextProps, TProps prevProps) areStatePropsEqual,
+  bool Function(TProps nextProps, TProps prevProps) areMergedPropsEqual,
+  Context context,
+  bool pure = true,
+  bool forwardRef = false,
+}) {
+  // Because of the complex relationship between actions and state, it should be
+  // enforced that a consumer cannot set both `...toProps` and the `withOwnProps`
+  // variants.
+  //
+  // Down stream, `...toProps` is defaulted to in this situation anyway, but at this level,
+  // allowing both can cause unnecessary complexity in decision logic.
+  assert((mapStateToProps == null || mapStateToPropsWithOwnProps == null) &&
+      (mapActionsToProps == null || mapActionsToPropsWithOwnProps == null));
```

Comment:
This seems like it should be an `ArgumentError` instead of an assert, since the consumer definitely did something wrong if they're passing in both